centos-ci/scripts/gluster-kubernetes.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+# if anything fails, we'll abort
+set -e
+
+# TODO: disable debugging
+set -x
+
+# we get the code from git and configure VMs with Ansible
+yum -y install git ansible
+
+# enable the SCL repository for Vagrant
+yum -y install centos-release-scl
+
+# install Vagrant with QEMU
+#
+# WARNING: adding sclo-vagrant1-vagrant on the "yum install" command makes it
+#          work fine. Without sclo-vagrant1-vagrant the following error occurs
+#          and starting the VMs fails:
+#
+#    Call to virDomainCreateWithFlags failed: the CPU is incompatible with host
+#    CPU: Host CPU does not provide required features: svm
+#
+yum -y install qemu-kvm sclo-vagrant1-vagrant sclo-vagrant1-vagrant-libvirt \
+               qemu-kvm-tools qemu-img
+
+# Vagrant needs libvirtd running
+systemctl start libvirtd
+
+# Log the virsh capabilites so that we know the
+# environment in case something goes wrong.
+virsh capabilities
+
+git clone https://github.com/gluster/gluster-kubernetes.git
+pushd gluster-kubernetes
+
+# by default we clone the master branch, but maybe this was triggered through a PR?
+if [ -n "${ghprbPullId}" ]
+then
+	git fetch origin pull/${ghprbPullId}/head:pr_${ghprbPullId}
+	git checkout pr_${ghprbPullId}
+	
+	# Now rebase on top of master
+	git rebase master
```

Comment:
Why do a rebase? You should build off what the pull request gives (useful if master is broken, right now, for instance)

Reply:
gluster-kubernetes doesn't allow merge commits, it's rebase-only. At least that's what I imagine is going on here.

Reply:
It also just came from the Heketi job, most of what is done there is done the same here.

Reply:
It makes sense to me because this job is supposed to be the final check before merging, so it should verify that the PR works against the merge target, in this case master. Additionally, with this testing in place we will have a greater guarantee that master won't be broken. :)

Reply:
It makes perfect sense to do a rebase on current master for a PR.
Or rather it even does not make sense not to.
Because that's the result of the push that would happen.

Reply:
If you're aware of the fallacies in doing this, I'm perfectly happy to merge.

Reply:
...apparently I'm not. What fallacies?

Reply:
If the successful test run does not trigger a merge, then between the test result and the later merge, master might have already changed again, which might have invalidated the test result. This is the only thing that I can think of that @nigelbabu can have meant.

The ideal thing in order to achieve correctness would be to hook one more test run as a gate to the merge button. But since we don't have that, doing a rebase on a test run at least makes it less likely to hit the problem. Furthermore our project does not show super high activity, which also reduces the chances.